### PR TITLE
DMP-2764 Delete event mapping

### PR DESCRIPTION
**Pull request by DavidTMann**
https://tools.hmcts.net/jira/browse/DMP-2764

- [X] commit messages are meaningful and follow good commit message guidelines
- [X] README and other documentation has been updated / added (if needed)
- [X] tests have been updated / new tests has been added (if needed)
- [ ] Does this PR introduce a breaking change

### CODE REVIEW

File: cypress/e2e/functional/admin-portal/event-mappings-spec.cy.js [hunk 1]
```diff
@@ -197,9 +197,14 @@ describe('Admin - Event mappings screen', () => {
...
+      cy.get('.delete-link').contains('Delete event mapping').click();
+      cy.get('.govuk-heading-l').contains('Are you sure want to delete this event mapping?');
 
-      //TBD
+      cy.get('app-data-table').contains('1010');
+      cy.get('app-data-table').contains('Changed name test');
+
+      cy.get('.govuk-button--warning').contains('Yes - delete').click();
+      cy.contains('Event mapping deleted').should('be.visible');
     });
```
Comment (davet1985) [inline cypress/e2e/functional/admin-portal/event-mappings-spec.cy.js:208]: Is there a test for the "You cannot delete this event mapping" flow?

File: src/app/admin/components/system-configuration/event-mappings/delete-event-mapping/delete-event-mapping.component.ts [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    TableRowTemplateDirective,
+    GovukHeadingComponent,
+  ],
+})
+export class DeleteEventMappingComponent {
+  eventMappingsService = inject(EventMappingsService);
+  route = inject(ActivatedRoute);
+  router = inject(Router);
+
+  id = +this.route.snapshot.params.id;
```
Comment (davet1985) [inline src/app/admin/components/system-configuration/event-mappings/delete-event-mapping/delete-event-mapping.component.ts:31]: Should the `+` be there?

**Review by davet1985 — COMMENTED** [review]
Couple of questions

File: cypress/e2e/functional/admin-portal/event-mappings-spec.cy.js (hunk 1 above)
Comment (DavidTMann) [reply cypress/e2e/functional/admin-portal/event-mappings-spec.cy.js:208]: Good spot - I will get this added

File: src/app/admin/components/system-configuration/event-mappings/delete-event-mapping/delete-event-mapping.component.ts (hunk 2 above)
Comment (DavidTMann) [reply src/app/admin/components/system-configuration/event-mappings/delete-event-mapping/delete-event-mapping.component.ts:31]: The + unary operator will convert the query params from string to the expected number type

Comment (davet1985) [reply src/app/admin/components/system-configuration/event-mappings/delete-event-mapping/delete-event-mapping.component.ts:31]: Didn't know that trick, nice

**Review by davet1985 — APPROVED** [review]

**Pull request merged**